src/database/utilities/createRecord.js

```diff
@@ -426,6 +426,13 @@ const createNumberSequence = (database, sequenceKey) =>
  * @param  {number}          number
  */
 const createNumberToReuse = (database, numberSequence, number) => {
+  const alreadyReusing = database
```

Comment:
Hmm, might be clearer to:

const `isNumberReused = database.objects(...).query(...).length > 0;`